test/infrastructure/test-extension/Dockerfile

```diff
@@ -0,0 +1,83 @@
+# syntax=docker/dockerfile:1.4
```

Comment:
I would prefer following the pattern of core CAPI and putting this all below test/extension

This is not an infra provider
(fine to align in a follow-up PR to reduce churn during reviews here)

Reply:
Okay not much churn (or at least easy churn). Thx for moving!